# HG changeset patch
# Parent 6f11186849a6ea05e55a15945a8fd455633add3c
add new API to get the pointer to the current clocksource

diff --git a/include/linux/clocksource.h b/include/linux/clocksource.h
--- a/include/linux/clocksource.h
+++ b/include/linux/clocksource.h
@@ -287,6 +287,7 @@ extern void clocksource_suspend(void);
 extern void clocksource_resume(void);
 extern struct clocksource * __init clocksource_default_clock(void);
 extern void clocksource_mark_unstable(struct clocksource *cs);
+extern struct clocksource * clocksource_get_current(void);
 
 extern void
 clocks_calc_mult_shift(u32 *mult, u32 *shift, u32 from, u32 to, u32 minsec);
diff --git a/kernel/time/clocksource.c b/kernel/time/clocksource.c
--- a/kernel/time/clocksource.c
+++ b/kernel/time/clocksource.c
@@ -786,6 +786,17 @@ void clocksource_unregister(struct clock
 }
 EXPORT_SYMBOL(clocksource_unregister);
 
+/**
+ * clocksource_get_current - return the pointer to the current clocksource
+ * 
+ */
+struct clocksource * clocksource_get_current(void)
+{
+	return curr_clocksource;
+}
+EXPORT_SYMBOL(clocksource_get_current);
+
+
 #ifdef CONFIG_SYSFS
 /**
  * sysfs_show_current_clocksources - sysfs interface for current clocksource
